providers/intercom/objectNames.go

```diff
@@ -0,0 +1,26 @@
+package intercom
+
+import (
+	"github.com/amp-labs/connectors/common/handy"
+	"github.com/amp-labs/connectors/providers/intercom/metadata"
+)
+
+// Supported object names can be found under schemas.json.
+var supportedObjectsByRead = handy.NewSet( //nolint:gochecknoglobals
+	metadata.Schemas.GetObjectNames(),
+)
+
+// ObjectNameToResponseField maps ObjectName to the response field name which contains that object.
+var ObjectNameToResponseField = handy.NewDefaultMap(map[string]string{ //nolint:gochecknoglobals
```

Comment:
Shouldn't we use this in the `intercom/read` like we are doing for zendesk?

Reply:
That was my initial intent. Any future connector that will have unique mapping between 2 concepts would do just like Zendesk handles it. But :smile: intercom always returns `type` field. Which tells us where the array is stored. This is convinent and I prefer programatic solution over hard coded as this requires no maintances.

But we could remove this code https://github.com/amp-labs/connectors/blob/3c4deeb3cf1f01e97b9f836b90bdb7d96c7c255e/providers/intercom/parse.go#L89 and fallback to `ObjectNameToResponseField` resolution. Even if provider API gets new endpoint that this programatic `type` inference will be able to catch our `ListObjectMetadata` would still not know about this. Since both Read and Metadata are so connected we can't harvest much benefit from programatic `type` inference, because ListObjectMetadata holds us back. 

@RajatPawar what is your opinion on this? Which approach do you prefer?

Reply:
Sure, if we can do it programmatically, that's always easier on the soul :) Can you just add a comment here that tells the reader about the context you added in here? Something like `Intercom returns a type field which tells us where the response array is located, which is why we do not need to hardcode the mapping. If we need to override this at any point, we can add the mapping here.` (please add additional context if missing)